src/components/user-form/AddUserForm.js

```diff
@@ -1,103 +1,154 @@
+        this.emailNameRef.current.value = '';
+        this.addressRef.current.value = '';
+    };
+
+
+    focusInput = () => {
+        console.log('focused');
+        this.lastNameRef.current.focus()
+    };
+
+    render() {
+        return (
+            <form onSubmit={this.onSubmit}>
+                {!!this.state.warning && <div>{this.state.warning}</div>}
+                <div className="form-group">
+                    <label htmlFor="firstNameInput">First name:</label>
+                    <input
+                        ref={this.firstNameRef}
+                        type="text"
+                        className="form-control"
+                        id="firstNameInput"
+                        placeholder="Example input"
+
+                    />
+                </div>
+
+                <div className="form-group">
+                    <label htmlFor="lastNameInput">Last name:</label>
+                    <input
+                        ref={this.lastNameRef}
+                        type="text"
+                        className="form-control"
+                        id="lastNameInput"
+                        placeholder="Example input"
+
+                    />
+                </div>
+
+                <div className="form-group">
+                    <label htmlFor="lastNameInput">Email:</label>
+                    <input
+                        ref={this.emailNameRef}
+                        type="text"
+                        className="form-control"
+                        id="lastNameInput"
+                        placeholder="Example input"
+                    />
+                </div>
+
+                {/*
+        done_todo 1: добавить инпут для ввода адресса
               передать ему атрибут под название ref наш созданный в строке 8 ref
         */}
 
-        <button type='submit' className="btn btn-primary m-2">Add </button>
-        <button type='button' onClick={this.focusInput}>focus</button>
-
-      {/*
-      todo 3: добавить кнопку, которая по нажатию будет будет вызывать метод onReset
+                <div className="form-group">
+                    <label htmlFor="lastNameInput">Address:</label>
+                    <input
+                        ref={this.addressRef}
+                        type="text"
+                        className="form-control"
+                        id="lastNameInput"
+                        placeholder="Example input"
+                    />
+                </div>
+
+                <button type='submit' className="btn btn-primary m-2">Add</button>
+                <button type='button' className="btn btn-secondary m-2" onClick={this.focusInput}>focus</button>
+
+                {/*
+      done_todo 3: добавить кнопку, которая по нажатию будет будет вызывать метод onReset
       */}
-      </form>
-    );
-  }
+                <button type="reset" className="btn btn-secondary m-2" onClick={this.onReset}>Reset</button>
```

Comment:
you can either use `type="button" onClick={this.onReset}` or just `type="reset"`
I guess both should work in the same way
Not a mistake, just take into account for the next time